src/httpclient.cc

```diff
@@ -0,0 +1,158 @@
+#include "httpclient.h"
+
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include "logger.h"
+
+/*****************************************************************************/
+/**
+ * \par Description:
+ *    A writeback handler for the curl library. I handles writing response
+ *    data from curl into a string.
+ *    https://curl.haxx.se/libcurl/c/CURLOPT_WRITEFUNCTION.html
+ *
+ */
+static size_t writeString(void* contents, size_t size, size_t nmemb,
+                          void* userp) {
+  // clear the provided string
+  ((std::string*)userp)->clear();
+
+  // append the writeback data to the provided string
+  ((std::string*)userp)->append((char*)contents, size * nmemb);
+
+  // return size of written data
+  return size * nmemb;
+}
+
+static size_t writeFile(void* contents, size_t size, size_t nmemb, FILE* fp) {
+  size_t written = fwrite(contents, size, nmemb, fp);
+  return written;
+}
+
+HttpClient::HttpClient() {
+  curl_global_init(CURL_GLOBAL_ALL);
```

Comment:
Yes, according to the documentation multiple calls of curl_global_init have the same effect as one call.